Sources/DistributedActors/Plugins/ClusterSingleton/ClusterSingleton.swift

```diff
@@ -271,21 +282,85 @@ internal distributed actor ClusterSingletonBoss<Act: DistributedActor>: _Cluster
             }
         }
     }
+
+    actor AllocationTracker {
+        /// The concrete distributed actor instance (the "singleton") if this node is indeed hosting it,
+        /// or nil otherwise - meaning that the singleton instance is actually located on another member.
+        var singleton: Act? {
+            didSet {
+                switch self.singleton {
+                case .some:
+                    self.status = .allocated
+                    self.timeoutTask?.cancel()
+                    self.timeoutTask = nil
+                case .none:
+                    self.status = .pending
+                    if self.timeoutTask == nil {
+                        Task {
+                            await self.startTimeoutTask()
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
okey good -- why as separate actor though, the boss could do this inside it already or do I miss something?

We'll get an extra actor hop when interacting with this one and issuing calls, preferably we wouldn't hm

Reply:
amended https://github.com/apple/swift-distributed-actors/pull/980/commits/3b882d62eb53c543108e291f2ffdab95105224de

Reply:
Great thanks!